### ISSUE TITLE
0 0 0 * * seems to be valid!

### ISSUE THREAD

**Issue by BentCoder** [mtdowling/cron-expression#120]
Hi,

Unfortunately code below passes! Date cannot be **0**.

Regards

**Code**

```
$isValid = VendorCronExpression::isValidExpression('0 0 0 * *');
echo 'TYPE:'.gettype($isValid);
echo PHP_EOL;
echo 'VALUE:'.$isValid;
```

**Result**

```
TYPE:boolean
VALUE:1
```

**Comment by BentCoder** [comment]
I guess you have a problem with https://github.com/mtdowling/cron-expression/blob/30a65ae8bcda39e11e11b97c1c126284e3f8ea8a/src/Cron/DayOfMonthField.php#L101

**Comment by dragonmantank** [comment]
Yup, you are correct. DayOfMonth should be 1-31, not 0-31. I'll make sure this gets fixed by the next release.

**Comment by dragonmantank** [comment]
This is finally corrected in dda25ed935212b6dbef0c89116bcec01c9c262f3, and will be part of the next release. Turns out that regex let a lot of things through it should not have.
